lua_examples/luaOTA/luaOTAserver.lua

```diff
@@ -0,0 +1,240 @@
+
+
+  while true do
+    local esp = server:accept()           -- wait for ESP connection 
+    esp:settimeout(ESPtimeout)            -- set session timeout
+    -- receive the opening request 
+    local config = receive_and_parse(esp)
+    if config and config.a == "HI" then 
+      print ("Processing provision check from ESP-"..config.id)
+      local inventory, fingerprint = get_inventory(src_dir, config.id)
+      -- Process the ESP request
+      if config.chk and config.chk == fingerprint then
+        send_command(esp, {r = "OK!"})       -- no update so send_command with OK
+        esp:receive("*l")                    -- dummy receive to allow client to close
+      else
+        local status, msg = pcall(provision, esp, config, inventory, fingerprint)
+        if not status then print (msg) end
+  
+    end
+    end
+    pcall(esp.close, esp)
+    print ("Provisioning complete")
+  end
+end
+
+-- Local Function Implementations ------------------------------------------------------
+
+-- Enumerate the sources directory and load the relevent inventory
+------------------------------------------------------------------
+get_inventory = function(dir, cpuid)
+  if (not dir or lfs.attributes(dir).mode ~= "directory") then
+    error("Cannot open directory, aborting %s" % arg[0], 0)
+  end
+
+  -- Load the CPU's (or the default) inventory 
+  local invtype, inventory = "custom", read_file("%s/ESP-%s.json" % {dir, cpuid})  
+  if not inventory then
+    invtype, inventory = "default", read_file(dir .. "/default.json")
+  end
+
+  -- tolerate and remove whitespace formatting, then decode
+  inventory = (inventory or ""):gsub("[ \t]*\n[ \t]*","")
+  inventory = inventory:gsub("[ \t]*:[ \t]*",":")
+  local ok; ok,inventory = pcall(json.decode, inventory)
+  if ok and inventory.files then
+    print( "Loading %s inventory for ESP-%s" % {invtype, cpuid})
+  else
+    error( "Invalid inventory for %s :%s" % {cpuid,inventory}, 0)
+  end
+
+  -- Calculate the current fingerprint of the inventory
+  local fp,f = {},inventory.files
+  for i= 1,#f do
+    local name, fullname = f[i], "%s/%s" % {dir, f[i]}
+    local fa = lfs.attributes(fullname)
+
+    assert(fa, "File %s is required but not in sources directory" % name)
+    fp[#fp+1] = name .. ":" .. fa.modification
+    f[i] = {name = name, mtime = fa.modification, 
+            size = fa.size, content = read_file(fullname) }
+    assert (f[i].size == #(f[i].content or ''), "File %s unreadable" % name )
+  end
+
+  assert (#f == #fp, "Aborting provisioning die to missing fies",0)
+  return inventory, md5.sumhexa(concat(fp,":"))
+end
+
+
+-- Encode a response buff, add a signature and any optional buffer
+------------------------------------------------------------------
+send_command = function(esp, resp, buffer)
+  if type(buffer) == "string" then 
+    resp.data = #buffer 
+  else 
+    buffer = ''
+  end
+  local rec = json.encode(resp)
+  rec = rec .. md5.sumhexa(secret .. rec):sub(-6) .."\n"
```

Comment:
'@nwf Nathaniel, thanks for this.  You are right, it is a more elegant approach. I will make these changes tomorrow and push them through once I am sure that I've got this implemented and tested correctly. :smile: